primitives/src/traits/dispute_api.rs

```diff
@@ -51,4 +51,65 @@ pub trait DisputeApi {
         market_id: &Self::MarketId,
         market: &Market<Self::AccountId, Self::BlockNumber, Self::Moment>,
     ) -> Result<Option<OutcomeReport>, DispatchError>;
+
+    /// Query the future resolution block of a disputed market.
+    /// **May** assume that `market.dispute_mechanism` refers to the calling dispute API.
+    ///
+    /// # Returns
+    ///
+    /// Returns the future resolution block if available, otherwise `None`.
+    fn get_auto_resolve(
+        disputes: &[MarketDispute<Self::AccountId, Self::BlockNumber>],
+        market_id: &Self::MarketId,
+        market: &Market<Self::AccountId, Self::BlockNumber, Self::Moment>,
+    ) -> Result<Option<Self::BlockNumber>, DispatchError>;
+
+    /// Query if the dispute mechanism failed for a dispute market.
+    /// **May** assume that `market.dispute_mechanism` refers to the calling dispute API.
+    ///
+    /// # Returns
+    ///
+    /// Returns `true` if the dispute mechanism failed. Otherwise `false`.
+    fn is_fail(
+        disputes: &[MarketDispute<Self::AccountId, Self::BlockNumber>],
+        market_id: &Self::MarketId,
+        market: &Market<Self::AccountId, Self::BlockNumber, Self::Moment>,
+    ) -> Result<bool, DispatchError>;
+}
+
+pub trait DisputeResolutionApi {
+    type AccountId;
+    type BlockNumber;
+    type MarketId;
+    type Moment;
+
+    /// Resolve a market.
+    ///
+    /// **Should** only be called if the market dispute
```

Comment:
Good question. I'm okay either way, but an error might be cleaner than "hoping" that the caller never messes up.

Reply:
Honestly, we could also remove `resolve` completely, since we don't use it at the moment 😄 . It is just, that we only have current scenarios, in which markets should resolve in the future.

But I can imagine, that this function is pretty useful and logical. I used it first, when I restructured `authorized`. The idea was to resolve the market directly, when the authority made its report.

It is up to you. Do we want to remove this API method?

Reply:
Malte suggested to leave it in, because it's useful and a common thing to resolve a market directly.